### PR TITLE
Add jth tests

### PR DESCRIPTION
**Pull request by tmigot**
Add tests related to the recent `jth_hess`, `jth_hess_coord` and `jth_hprod` functions.

[will break ADNLPModels]

### CODE REVIEW

**Comment by tmigot** [comment]
Breakage fixes:
* LLSModels [here](https://github.com/tmigot/LLSModels.jl/tree/exluce-jth_functions)
* ADNLPModels [add implementation of the new functions](https://github.com/tmigot/ADNLPModels.jl/tree/exlude-jth_functions)

File: src/nlp/consistency.jl [hunk 1]
```diff
@@ -437,6 +437,64 @@ function consistent_functions(nlps; rtol=1.0e-8, exclude=[])
       end
     end
 
+    if !(jth_hess_coord in exclude)
+      k = m
```
Comment (abelsiqueira) [inline src/nlp/consistency.jl:441]: Was this going to be a `for`?

**Review by abelsiqueira — CHANGES_REQUESTED** [review]
Thanks. I just didn't understood those fixed `k` values.

File: src/nlp/consistency.jl (hunk 1 above)
Comment (tmigot) [reply src/nlp/consistency.jl:441]: It is true, they are not so useful.
Making another `for` and testing all the constraints hessian might become very expansive, what do you think?

Comment (abelsiqueira) [reply src/nlp/consistency.jl:441]: I'm happy with a single value being tested. @dpo, opinions?

**Comment by abelsiqueira** [comment]
I completely forgot the situation of this PR.

**Comment by tmigot** [comment]
@abelsiqueira I rebased and it seems good to go

**Comment by abelsiqueira** [comment]
Thanks

**Pull request merged**